engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -784,10 +784,15 @@ public void monitorHeartbeat(boolean waitForRecovery)
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
-          if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
+          if (sca.lastHeartbeatMillis != -1 && (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis)) {
             if (!isApplicationIdle()) {
```

Comment:
@tweise @PramodSSImmaneni  This does not look right to me. Should not a container be killed even when an application is idle when the container misses a heartbeat?